src/lib/import_export/csv_meta.cpp

```diff
@@ -0,0 +1,86 @@
+#include "csv_meta.hpp"
+
+#include <fstream>
+#include <string>
+#include <vector>
+
+#include "utils/assert.hpp"
+
+namespace opossum {
+
+CsvMeta process_csv_meta_file(const std::string& filename) {
+  std::ifstream metafile{filename};
+  Assert(metafile.good(), "Meta file does not exist: " + filename);
+  nlohmann::json meta_json;
+  metafile >> meta_json;
+  CsvMeta meta = meta_json;
+  return meta;
+}
+
+void assign_if_exists(char& value, const nlohmann::json& json_object, const std::string& key) {
+  if (json_object.find(key) != json_object.end()) {
+    std::string character = json_object.at(key);
+    Assert(character.size() == 1, "CSV meta file config: Character specifications can only be a single character.");
+    value = character[0];
+  }
+}
+
+void assign_if_exists(bool& value, const nlohmann::json& json_object, const std::string& key) {
+  if (json_object.find(key) != json_object.end()) {
+    value = json_object.at(key);
+  }
+}
+
+void from_json(const nlohmann::json& json, CsvMeta& meta) {
+  meta.chunk_size = json.at("chunk_size");
+
+  // Apply only parts of the ParseConfig that are provided, use default values otherwise
+  ParseConfig config{};
+  if (json.find("config") != json.end()) {
+    Assert(json.at("config").is_object(), "CSV meta file,\"Config\" field has to be a json object.");
+    nlohmann::json config_json = json.at("config");
+    assign_if_exists(config.delimiter, config_json, "delimiter");
+    assign_if_exists(config.separator, config_json, "separator");
+    assign_if_exists(config.quote, config_json, "quote");
+    assign_if_exists(config.escape, config_json, "escape");
+    assign_if_exists(config.delimiter_escape, config_json, "delimiter_escape");
+    assign_if_exists(config.reject_quoted_nonstrings, config_json, "reject_quoted_nonstrings");
+    assign_if_exists(config.rfc_mode, config_json, "rfc_mode");
+  }
+
+  if (json.find("columns") != json.end()) {
+    Assert(json.at("columns").is_array(), "CSV meta file,\"Columns\" field has to be a json array.");
+    std::vector<ColumnMeta> columns;
+    for (auto column : json.at("columns")) {
+      ColumnMeta column_meta;
+      column_meta.name = column.at("name");
+      column_meta.type = column.at("type");
+      assign_if_exists(column_meta.nullable, column, "nullable");
+      columns.push_back(column_meta);
+    }
+    meta.columns = columns;
+  }
+
+  meta.config = config;
+}
+
+void to_json(nlohmann::json& json, const CsvMeta& meta) {
```

Comment:
All these methods are currently global functions inside the `opossum` namespace, which I think should be avoided. Either create a namespace for them, e.g., `opossum::csv_import` or assign them to a `class` if appropriate.

Reply:
Also, some explanatory comments for these functions would be great

Reply:
I will add some comments to that.

Reply:
To the namespace issue, I could do that I think, although then I'd *have to* put the struct (`CsvMeta`) and everything else into that namespace as well. The `from_json` and `to_json` methods are used internally to make the assignments (`json = CsvMeta` and the other way) work, and therefore the type and the function has to be in the same namespace.

Reply:
as long as we do not have a global `opossum::to_json` method, I'm okay with the solution I think. maybe get some input from the other people on how to handle this.

Reply:
All of these methods are in the cpp file and are never exported. Since they never leave this file, I would be fine with it.

Reply:
So just to clarify, what is the final say here now? Leave it as it is, or move everything in `csv_meta.*` to `opossum::csv::` namespace?

Reply:
hmmm I didn't think about the function being in the `cpp` only. I guess I'm fine with that. Just leave it as it is